src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -102,50 +105,53 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource)(implicit ec: E
-            if (!subFields(1).isInstanceOf[JsString]) throw ValidationException("Configuration methodRepoMethod must include a 'methodName' key with a string value")
-            if (!subFields(2).isInstanceOf[JsNumber]) throw ValidationException("Configuration methodRepoMethod must include a 'methodVersion' key with a JSNumber value")
-          }
+  private def checkValidPayload[T](agoraEntity: AgoraEntity)(op: => ReadWriteAction[T]): ReadWriteAction[T] = {
+    agoraEntity.payload match {
+      case None =>
+        DBIO.failed(ValidationException(s"Agora entity $agoraEntity has no payload."))
+      case Some(payload) =>
+        val payloadOK = agoraEntity.entityType match {
+          case Some(AgoraEntityType.Task) =>
+            WdlNamespace.loadUsingSource(payload, None, Option(Seq(httpResolver(_))))
+          // NOTE: Still not validating existence of docker images.
+          // namespace.tasks.foreach { validateDockerImage }
+
+          case Some(AgoraEntityType.Workflow) =>
+            WdlNamespaceWithWorkflow.load(payload, Seq(httpResolver(_)))
+          // NOTE: Still not validating existence of docker images.
+          //namespace.tasks.foreach { validateDockerImage }
+
+          case Some(AgoraEntityType.Configuration) =>
+            Try {
+              val json = payload.parseJson
+              val fields = json.asJsObject.getFields("methodRepoMethod")
+              if (fields.size != 1) throw ValidationException("Configuration payload must define at least one field named 'methodRepoMethod'.")
+
+              val subFields = fields.head.asJsObject.getFields("methodNamespace", "methodName", "methodVersion")
+              if (!subFields(0).isInstanceOf[JsString]) throw ValidationException("Configuration methodRepoMethod must include a 'methodNamespace' key with a string value")
+              if (!subFields(1).isInstanceOf[JsString]) throw ValidationException("Configuration methodRepoMethod must include a 'methodName' key with a string value")
+              if (!subFields(2).isInstanceOf[JsNumber]) throw ValidationException("Configuration methodRepoMethod must include a 'methodVersion' key with a JSNumber value")
+            }
 
-        case _ => //hello "shouldn't get here" my old friend
-          Failure(ValidationException(s"AgoraEntity $agoraEntity has no type!"))
-      }
+          case _ => //hello "shouldn't get here" my old friend
+            Failure(ValidationException(s"AgoraEntity $agoraEntity has no type!"))
+        }
 
-    payloadOK match {
-      case Success(_) => op
-      case Failure(e: SyntaxError) =>
-        DBIO.failed(ValidationException(e.getMessage, e.getCause))
-      case Failure(regret) =>
-        DBIO.failed(regret)
+        payloadOK match {
+          case Success(_) => op
+          case Failure(e @ (_: SyntaxError | _: WdlValidationException)) =>
+            DBIO.failed(ValidationException(s"$errorMessagePrefix ${e.getMessage}", e.getCause))
```

Comment:
`errorMessagePrefix` is used exactly once, IMO putting it in a variable is not worth it (and adds a step when locating the source of the error)